starknet_py/tests/e2e/docs/guide/test_using_existing_contracts.py

```diff
@@ -26,40 +25,24 @@
 
 
 @pytest.mark.asyncio
-async def test_using_existing_contracts(
-    gateway_client, gateway_account_client, erc20_contract
-):
+async def test_using_existing_contracts(account_client, erc20_contract):
     # pylint: disable=import-outside-toplevel,too-many-locals,unused-variable
     # docs: start
-    from starknet_py.net.gateway_client import GatewayClient
     from starknet_py.contract import Contract
-    from starknet_py.net.networks import TESTNET
-    from starknet_py.net.models import StarknetChainId
 
     address = "0x00178130dd6286a9a0e031e4c73b2bd04ffa92804264a25c1c08c1612559f458"
-    client = GatewayClient(TESTNET)
-    # docs: end
-    client = gateway_client
-    # docs: start
 
-    contract = Contract(address=address, abi=abi, client=gateway_client)
+    contract = Contract(address=address, abi=abi, client=account_client)
```

Comment:
Lets add a comment that this is a preferred method.

Reply:
And let's include info that this way we save ourselves additional request to fetch the ABI.

Reply:
Isn't it a bit too technical for a (rather simple) guide? I mean, we can include it but I wonder if we should.

Reply:
This gives user a reason for doing it **_properly_**